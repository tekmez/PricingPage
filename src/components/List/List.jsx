import React from 'react'
import { FlatList, View } from 'react-native'
import styles from './List.style'
import ListItem from '../ListItem'
const List = ({ data }) => {
  return (
    <FlatList
      data={data}
      testID="list"
      renderItem={({ item }) => (
        <ListItem
          icon={item.icon}
          title={item.title}
          description={item.description}
        />
      )}
      keyExtractor={(item) => item.id}
      style={styles.container}
      ItemSeparatorComponent={
        <View
          testID="list-separator"
          style={{
            height: 1,
            backgroundColor: '#E7EAF4',
            marginHorizontal: 10,
          }}
        />
      }
    />
  )
}

export default List
