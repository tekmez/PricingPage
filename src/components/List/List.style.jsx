import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
})

export default styles
