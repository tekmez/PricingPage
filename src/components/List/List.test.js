import React from 'react'
import { render } from '@testing-library/react-native'
import List from './List'
const mockData = [
  { id: 1, icon: 'icon1', title: 'Title 1', description: 'Description 1' },
  { id: 2, icon: 'icon2', title: 'Title 2', description: 'Description 2' },
  { id: 3, icon: 'icon3', title: 'Title 3', description: 'Description 3' },
]
describe('List', () => {
  it('should match snapshot', () => {
    const comp = render(<List />).toJSON()
    expect(comp).toMatchSnapshot()
  })
  it('renders correctly with mock data', () => {
    const { getByTestId, getAllByTestId } = render(<List data={mockData} />)
    const listComponent = getByTestId('list')
    const listItemComponents = getAllByTestId('list-item-container')
    expect(listComponent).toBeTruthy()
    expect(listItemComponents.length).toBe(mockData.length)
  })

  it('renders ListItem components with correct data', () => {
    const { getAllByTestId } = render(<List data={mockData} />)
    const listItemComponents = getAllByTestId('list-item-container')

    const iconComponents = getAllByTestId(`list-item-icon`)
    const titleComponents = getAllByTestId(`list-item-title`)
    const descriptionComponents = getAllByTestId(`list-item-description`)

    listItemComponents.forEach((item, index) => {
      const icon = iconComponents[index].children[0]
      const title = titleComponents[index].children[0]
      const description = descriptionComponents[index].children[0]

      expect(icon).toBe(mockData[index].icon)
      expect(title).toBe(mockData[index].title)
      expect(description).toBe(mockData[index].description)
    })
  })

  it('renders ItemSeparatorComponent between ListItem components', () => {
    const { getAllByTestId } = render(<List data={mockData} />)
    const separatorComponents = getAllByTestId('list-separator')

    expect(separatorComponents.length).toBe(mockData.length - 1)
  })
})
