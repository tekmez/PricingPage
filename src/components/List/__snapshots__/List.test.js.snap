// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<RCTScrollView
  ItemSeparatorComponent={
    <View
      style={
        {
          "backgroundColor": "#E7EAF4",
          "height": 1,
          "marginHorizontal": 10,
        }
      }
      testID="list-separator"
    />
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  style={
    {
      "backgroundColor": "#fff",
    }
  }
  testID="list"
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;
